de.marw.cdt.cmake.core/src/main/java/de/marw/cdt/cmake/core/internal/BuildscriptGenerator.java

```diff
@@ -431,6 +444,46 @@ private MultiStatus invokeCMake(IPath srcDir, IPath buildDir, IConsole console)
       return new MultiStatus(CdtPlugin.PLUGIN_ID, IStatus.ERROR, errMsg, null);
     }
   }
+  
+  
+  private void setupMakeTargets() {
+    IPath makefilePath = buildFolder.getLocation().append(getMakefileName());
```

Comment:
This will return 'rules.ninja' or other values if a generator other than Unix Makefiles is in effect. You jshould check, which generator is in effect, before. See CmakeGenerator.Ninja#getMakefileName()